sdcm/utils/quota.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import absolute_import, annotations
+
+import time
+import logging
+from sdcm.exceptions import QuotaConfigurationFailure
+
+LOGGER = logging.getLogger(__name__)
+
+
+def configure_quota_on_node_for_scylla_user(target_node):
+    """Configure quota with current disk usage size + 500MB for scylla user which uses /var/lib/scylla mount point"""
+    result = target_node.remoter.run("df -al | grep '/var/lib/scylla'")
+    used_space = int(result.stdout.split()[2])
+    LOGGER.debug('Currently used space is: {}K'.format(used_space))
+    quota_size = used_space + 500000
+    LOGGER.info(
+        "Configuring quota for scylla user on /var/lib/scylla with quota size: {}K on node {}".format(quota_size, target_node))
+    conf_quota_cmd = f"xfs_quota -x -c 'limit bsoft={quota_size}K bhard={quota_size}K scylla' /var/lib/scylla"
+    target_node.remoter.sudo(conf_quota_cmd, verbose=True, )
+
+
+def remove_quota_configuration_from_node(target_node):
+    LOGGER.debug('Remove quota configuration from user \'scylla\' by setting it to 0')
+    quota_remove_cmd = "xfs_quota -x -c 'limit bsoft=0 bhard=0 scylla' /var/lib/scylla"
+    target_node.remoter.sudo(quota_remove_cmd, verbose=True)
+
+
+def verify_quota_is_configured_for_scylla_user(target_node):
+    """Verifies quota ia actually configured for scylla user, but doesn't verify enforcement in ON"""
+    get_scylla_user_quota_value = 'xfs_quota -x -c "report -h" /var/lib/scylla | grep scylla | tail -1 | awk {\'print $4\'}'
+    LOGGER.info("Verifying quota is configured for scylla user")
+    quota_value = target_node.remoter.sudo(get_scylla_user_quota_value).stdout.rstrip()
+    if quota_value == "0":
+        LOGGER.error("Failed to configure quota on user")
+        raise QuotaConfigurationFailure(
+            "Quota was not configured for scylla user. The value for hard quota is: {}".format(quota_value))
+    LOGGER.info("Quota was configured for scylla user with value: {}".format(quota_value))
+
+
+def enable_quota_on_node(node):
+    LOGGER.debug('Enabling quota on node {}'.format(node.name))
+    change_default_grub_cmd = 'sed -i s\'/GRUB_CMDLINE_LINUX_DEFAULT="quiet splash"/' \
+                              'GRUB_CMDLINE_LINUX_DEFAULT="quiet splash rootflags=uquota,pquota"/\' ' \
+                              '/etc/default/grub'
+    mk_grub_cmd = 'grub-mkconfig -o /boot/grub/grub.cfg'
+    enable_uquota_on_mount_cmd = 'sed -i s\'/Options=noatime,discard/Options=noatime,discard,uquota/\' ' \
+                                 '/etc/systemd/system/var-lib-scylla.mount'
+
+    node.remoter.sudo(cmd=change_default_grub_cmd)
+    node.remoter.sudo(cmd=mk_grub_cmd)
+    node.remoter.sudo(cmd=enable_uquota_on_mount_cmd)
+    LOGGER.debug('Rebooting node: "{}"'.format(node.name))
+    node.reboot(hard=False)
+    LOGGER.debug("node reboot takes very long time and next methods fails. "
+                 "Adding sleep for 2 minutes here before wait_node_fully_start")
+    time.sleep(120)
```

Comment:
Shouldn't be like that, at the maximum you should do "wait_for" and increase timeout.